app/Cmd.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cmd (
+  CmdOpts (CmdOpts),
+  MongoOpts (MongoOpts)
+  ) where
+
+import           Data.Bifunctor
+import           Data.Maybe
+import           Data.Text        hiding (group)
+import           Database.MongoDB hiding (Group, group)
+import           Options
+
+-------------------------------------------------------------------------------------------------------------------------
+-- Global options
+-------------------------------------------------------------------------------------------------------------------------
+data CmdOpts = CmdOpts {
+  mongoOpts :: MongoOpts,
+  port      :: Int
+  } deriving (Eq, Show)
+
+instance Options CmdOpts where
+  defineOptions = pure CmdOpts
+                  <*> defineOptions
+                  <*> simpleOption "port" 3000 "Port on which to listen for incoming HTTP calls."
+
+-------------------------------------------------------------------------------------------------------------------------
+-- MongoDB options
+-------------------------------------------------------------------------------------------------------------------------
+
+data MongoOpts = MongoOpts {
+  mongoHost :: Host,
+  mongoDb   :: Text,
+  mongoCol  :: Text
+  } deriving (Eq, Show)
+
+mongoGroup :: Group
+mongoGroup = group "mongodb" "MongoDB options" "Show MongoDB specific options"
+
+optionType_host :: OptionType Host
+optionType_host = optionType "mongodb host" (host "localhost") parseHost showHostPort
+    where
+      parseHost = first (const "Invalid mongodb host") . readHostPortM
```

Comment:
When reporting error messages to the user, it's always good practice to show them what wasn't right:

```suggestion
      parseHost = first (\e ->  "Invalid mongodb host, got: " <> show e) . readHostPortM
```

Reply:
Agreed, it's just that `Options`' error messages are rubbish :)
That being said: is it really more common to use `<>` than  `++` when concatenating strings?

Reply:
I tend to use `++` when I want to monomorphize a polymorphic literal to `String`. With `OverloadedStrings`, the type of `"asdf" <> "bar" :: (IsString a, Monoid a) => a`, and GHC will say "Ambiguous type! I don't know what to pick for this." Whereas `"asdf" ++ "bar" :: [Char]` and GHC doesn't complain.

`<>` works for `Text`, `String`, `ByteString`, etc. and when the type is known/fixed from some other source, that lets me not have to worry about *which* type it actually is. I didn't know if the error messages were String, Text, or some other type, so I picked `<>` to operate on it.

Reply:
Should be fixed in 24e7bde2bcc268d64f3c32bbab00daa4b263af27.